uber/models.py

```diff
@@ -1057,12 +1057,10 @@ def _status_adjustments(self):
                            render('emails/reg_workflow/banned_attendee.txt', {'attendee': self}), model='n/a')
             except:
                 log.error('unable to send banned email about {}', self)
-        elif self.badge_status == c.NEW_STATUS and not self.placeholder and self.first_name:
-            if self.paid in [c.HAS_PAID, c.NEED_NOT_PAY] \
-                    or self.paid == c.PAID_BY_GROUP and self.group and not self.group.amount_unpaid:
-                self.badge_status = c.COMPLETED_STATUS
-        elif self.badge_status == c.INVALID_STATUS and self.admin_account:
-            self.session.delete(self.admin_account)
+        elif self.badge_status == c.NEW_STATUS and not self.placeholder and self.first_name \
+                and (self.paid in [c.HAS_PAID, c.NEED_NOT_PAY]
+                     or self.paid == c.PAID_BY_GROUP and self.group_id and not self.group.amount_unpaid):
+            self.badge_status = c.COMPLETED_STATUS
```

Comment:
Two changes here:

1) I eliminated the "delete the admin account of an attendee with an invalid badge" clause.  This seemed like a desirable feature at the time, but there are actually situations where we don't want this.  We've already run into situations where it's caused us issues - specifically, we have MIVS judges who are not attending MAGFest, and they need to keep their admin accounts intact.

2) I did a minor style refactor, since with the final if-statement gone, we were just saying

```python
if foo:
    if bar:
```

instead of

```python
if foo and bar:
```